### PR TITLE
Set the meta data before everything

### PR DESCRIPTION
**Pull request by nickvergessen**
If the meta data is not set in the beginning customisation can only be done on the final out put method instead of in all of the methods.

While thinking about this. Should we merge the function into the `createEMailTemplate` function and the constructor?

### CODE REVIEW

File: lib/private/Mail/Mailer.php [hunk 1]
```diff
@@ -89,21 +90,33 @@ public function createMessage() {
...
 	}
 
-	public function createEMailTemplate() {
+	/**
+	 * Creates a new email template object
+	 *
+	 * @param string $emailId
+	 * @param array $data
+	 * @return IEMailTemplate
+	 * @since 12.0.0
```
Comment (MorrisJobke) [inline lib/private/Mail/Mailer.php]: 12.0.3?

Comment (nickvergessen) [reply lib/private/Mail/Mailer.php]: See public interface

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Just to know: calendar and circles also need an update then as well. 🙈

**Review by rullzer — APPROVED** [review]

**Comment by rullzer** [comment]
Failing tests!

**Comment by MorrisJobke** [comment]
> Failing tests!

Fixed.

**Comment by nickvergessen** [comment]
> To github.com:nextcloud/server.git
 ! [rejected]              fix-mixup-of-id-and-name -> fix-mixup-of-id-and-name (fetch first)

seconds

**Comment by nickvergessen** [comment]
Backport in https://github.com/nextcloud/server/pull/6366

**Pull request merged**